### PR TITLE
Vagrantfile and miscellaneous fixes

### PR DESCRIPTION
**Pull request by tanabarr**
- add Vagrantfile for development purposes
- bump paket package versions
- fix order of project files so that entry point is last
- fix changelog rpmlint warning

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+  #
+  # Create a device-aggregator node
+  #
+  config.vm.define "device-aggregator", primary: true do |device_aggregator|
+    device_aggregator.vm.provider "virtualbox" do |v|
+      v.memory = 4096
+      v.name = "device-aggregator"
+
+      file_to_disk = './tmp/device_aggregator.vdi'
```
Comment (jgrund) [inline Vagrantfile]: You can remove all this extra disk stuff since the aggregator doesn't need it

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    device_aggregator.vm.provision "shell", inline: "cat >/root/.ssh/config<<__EOF
+Host testnode
+  Hostname #{$test_ip}
+  StrictHostKeyChecking no
+__EOF"
+
+    device_aggregator.vm.provision "shell", inline: <<-SHELL
+    rpm --import "http://keyserver.ubuntu.com/pks/lookup?op=get&search=0x3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF"
+    yum-config-manager --add-repo http://download.mono-project.com/repo/centos7/
+    wget https://bintray.com/intel-hpdd/intelhpdd-build/rpm -O /etc/yum.repos.d/bintray-intel-hpdd-intelhpdd-build.repo
```
Comment (jgrund) [inline Vagrantfile]: Can we remove this?

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    RPM_NAME=iml-device-aggregator-$PACKAGE_VERSION-1.el7.centos.x86_64.rpm
+    docker cp mock:/var/lib/mock/epel-7-x86_64/result/$RPM_NAME ./
+    yum install -y ./$RPM_NAME
+    SHELL
+  end
+
+  #
+  # Create a test node
+  #
+  config.vm.define "test", primary: false, autostart: false do |test|
```
Comment (jgrund) [inline Vagrantfile]: Do we need this extra node?

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    test.vm.provision "shell", inline: <<-SHELL
+rpm --import "http://keyserver.ubuntu.com/pks/lookup?op=get&search=0x3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF"
+yum-config-manager --add-repo http://download.mono-project.com/repo/centos7/
+yum install -y epel-release
+yum install -y centos-release-dotnet
+yum install -y nodejs mono-devel rh-dotnet20
+rm -rf /builddir
+cp -r /vagrant /builddir
+cd /builddir
+npm i --ignore-scripts
```
Comment (jgrund) [inline Vagrantfile]: no need for --ignore-scripts in this module, it doesn't compile native code.

File: device-aggregator.spec [hunk 5]
```diff
@@ -71,10 +71,15 @@ if [ $1 -eq 0 ] ; then
 fi
 
 %changelog
+* Wed Feb 28 2018 Tom Nabarro <tom.nabarro@outlook.com> - 1.0.1-1
+- include Vagrantfile and keys
+- fix project file order
+- fix rpmlint spec changelog warning
+
 * Mon Feb 26 2018 Tom Nabarro <tom.nabarro@outlook.com> - 1.0.0-1
```
Comment (jgrund) [inline device-aggregator.spec]: Can you change the above two addrs to intel ones?

File: Vagrantfile [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure("2") do |config|
+  config.vm.box = "manager-for-lustre/centos74-1708-base"
+  config.vm.synced_folder ".", "/vagrant", type: "virtualbox"
+
+  config.vm.provider "virtualbox" do |vb|
+    vb.memory = 2048
```
Comment (johnsonw) [inline Vagrantfile]: Any reason why 1024 isn't sufficient?

File: Vagrantfile [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure("2") do |config|
+  config.vm.box = "manager-for-lustre/centos74-1708-base"
+  config.vm.synced_folder ".", "/vagrant", type: "virtualbox"
```
Comment (johnsonw) [inline Vagrantfile]: Not worth changing but vagrant will select the best type for synced_folder if the type isn't specified. In the Vagrant files setup for the different versions of centos, for example, the type isn't specified.

File: Vagrantfile (hunk 6 above)
Comment (tanabarr) [reply Vagrantfile]: kept getting memory allocation errors while running on Windows/Cygwin, solved by increasing memory here^

File: Vagrantfile (hunk 7 above)
Comment (tanabarr) [reply Vagrantfile]: I've had trouble with not specifying type on Windows

**Pull request merged**